src/marqo/marqo_cloud_instance_mappings.py

```diff
@@ -67,9 +75,20 @@ def _refresh_urls(self, timeout=None):
             self.latest_index_mappings_refresh_timestamp = time.time()
 
     def index_http_error_handler(self, index_name: str, http_status: Optional[int] = None) -> None:
-        mq_logger.debug(f'Evicting index {index_name} from cache (if exists) due to error {http_status} and '
-                        f'refreshing index URL cache')
+        mq_logger.debug(f'Triggering cache refresh due to error on index {index_name}')
 
-        self._urls_mapping[IndexStatus.READY].pop(index_name, None)
+        self._refresh_urls_if_needed()
 
-        self._refresh_urls_if_needed(index_name)
+    def is_index_usage_allowed(self, index_name: str) -> bool:
+        """Checks the status of the index in self._urls_mapping.
+
+        Note that this method does not request a refresh of the mappings, so the result
+        of this method is a best effort.
+
+        If the index_name cannot be found in the self._urls_mapping, then False will be returned.
+        """
+
+        if index_name in self._urls_mapping[IndexStatus.READY]:
```

Comment:
Since we don't have periodical refreshes anymore, if a user relied on this and never tried any other operation on the index (which will trigger a cache refresh), they would get False indefinitely

Reply:
Yep. I mention that this is a best effort. This is intended to used to reflect the state of whether the client thinks it's OK to start using the index, rather than attempting to reflect the reality of the index state. 

Currently it is just used to help determine whether to skip an index's version check.  For the default mappings, it always returns True, as the client should always be allowed to check the index's version.

Let me know if you can think of a better abstraction. Perhaps we should just make it more specific and not attempt to be general `is_index_version_check_allowed()`?